estate/models/property.py

```diff
@@ -0,0 +1,110 @@
+
+from odoo import api, fields, models
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools import float_compare, float_is_zero
+
+
+class Property(models.Model):
+    _name = 'estate.property'
+    _description = "Estate property"
+    _order = 'id desc'
+
+    name = fields.Char(string="Title", required=True)
+    description = fields.Text()
+    postcode = fields.Char()
+    date_availability = fields.Date(
+        string="Available From",
+        default=lambda self: fields.Date.today() + relativedelta(months=3),
+        copy=False,
+    )
+    expected_price = fields.Float(required=True)
+    selling_price = fields.Float(readonly=True, copy=False)
+    bedrooms = fields.Integer(default=2)
+    living_area = fields.Integer(string="Living Area (sqm)")
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer(string="Garden Area (sqm)")
+    garden_orientation = fields.Selection(
+        selection=[('north', 'North'), ('south', 'South'), ('east', 'East'), ('west', 'West')]
+    )
+    state = fields.Selection(
+        default="new",
+        string="Status",
+        selection=[
+            ('new', 'New'),
+            ('offer_received', 'Offer Received'),
+            ('offer_accepted', 'Offer Accepted'),
+            ('sold', 'Sold'),
+            ('cancelled', 'Cancelled'),
+        ],
+        required=True,
+        copy=False,
+    )
+    active = fields.Boolean(default=True)
+    property_type_id = fields.Many2one('estate.property.type', string="Property Type")
+    partner_id = fields.Many2one('res.partner', string="Buyer", readonly=True)
+    user_id = fields.Many2one('res.users', string="Salesman", default=lambda self: self.env.user)
+    tag_ids = fields.Many2many('estate.property.tag', string="Property Tags")
+    offer_ids = fields.One2many('estate.property.offer', 'property_id', string="Offers")
+    total_area = fields.Integer(string="Total Area (sqm)", compute='_compute_total_area')
+    best_offer = fields.Float(default=0.0, compute="_compute_best_offer")
+
+    _expected_price_strictly_pos = models.Constraint(
+        "CHECK(expected_price > 0)", "The expected price must be strictly positive."
+    )
+    _selling_price_pos = models.Constraint(
+        "CHECK(selling_price >= 0)", "The selling price must be positive."
+    )
+
+    @api.depends('living_area', 'garden_area')
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+
+    @api.depends('offer_ids')
+    def _compute_best_offer(self):
+        for record in self:
+            record.best_offer = max(record.offer_ids.mapped('price')) if record.offer_ids else 0.0
+
+    @api.onchange('garden')
+    def _onchange_garden(self):
+        self.garden_area = 10 if self.garden else 0
+        self.garden_orientation = 'north' if self.garden else ''
```

Comment:
I would prefer this version as it might lead to some extensions in the future but not a required change so far
```suggestion
        if self.garden:
            self.garden_area = 10
            self.garden_orientation = 'north'
        if self.garden:
            self.garden_area = 0
            self.garden_orientation = False
```